packages/libs/error-stack/src/frame.rs

```diff
@@ -308,6 +344,88 @@ impl VTable {
     }
 }
 
+/// Stores a [`Box`] and a [`FrameKind`] by only occupying one pointer in size.
+///
+/// It's guaranteed, that a `TaggedBox` has the same size as `Box`.
+pub struct TaggedBox<T>(usize, PhantomData<Box<T>>);
+
+impl<T> TaggedBox<T> {
+    /// Mask for the pointer.
+    ///
+    /// For a given pointer width, this will be
+    ///  - 16 bit: `1111_1111_1111_1110`
+    ///  - 32 bit: `1111_1111_1111_1111_1111_1111_1111_1100`
+    ///  - 64 bit: `1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1000`
+    /// so the last bit will *always* be `0`.
+    const MASK: usize = !(mem::align_of::<*const T>() - 1);
+
+    /// Creates a new tagged pointer with the tag
+    ///
+    /// # Panics
+    ///
+    /// if the tag is too large to be stored next to a pointer
+    pub fn new(frame: T, kind: FrameKind) -> Self {
+        // Will only fail on 8-bit platforms which Rust currently is not supporting
+        assert!(
+            mem::align_of::<*const T>() >= 2,
+            "Tag can't be stored as tagged pointer"
+        );
+        let raw = Box::into_raw(Box::new(frame));
+
+        let tag = kind == FrameKind::Context;
+        // Store the tag in the last bit, due to alignment, this is 0 for 16-bit and higher
+        Self(raw as usize | usize::from(tag), PhantomData)
+    }
+
+    /// Returns the tag stored inside the pointer
+    pub const fn kind(&self) -> FrameKind {
+        if self.0 & 1 == 1 {
```

Comment:
Add a comment (similar to above) stating that we store the tag in the last bit with it being 1 if context and 0 if attachment.